distribution/std-lib/Standard/src/Table/Data/Table.enso

```diff
@@ -311,10 +315,69 @@ type Table
 
     ## UNSTABLE
 
-       Returns a table truncated to at most `max_rows` rows.
+       Returns a table containing the first `count` elements in this table.
+
+       Arguments:
+       - count: The number of elements to take from the start of this table.
+
+       If `this` has a number of rows in it less than `count`, the entire table
+       will be returned.
     take_start : Integer -> Table
-    take_start max_rows =
-        Table (this.java_table.slice 0 max_rows)
+    take_start count = Table (this.java_table.slice 0 count)
+
+    ## UNSTABLE
+
+       Returns a table containing the last `count` elements in this table.
+
+       Arguments:
+       - count: The number of elements to take from the end of this table.
+
+       If `this` has a number of rows in it less than `count`, the entire table
+       will be returned.
+    take_end : Integer -> Table
+    take_end count =
+        start_point = Math.max (this.row_count - count) 0
+        Table (this.java_table.slice start_point count)
+
+    ## UNSTABLE
+
+       Returns the first row in the table, if it exists.
+
+       If the table is empty, this method will return a dataflow error
+       containing `Nothing`.
+    first : Table ! Nothing
+    first =
+        table = this.take_start 1
+        if table.row_count != 1 then Error.throw Nothing else table
```

Comment:
Are you sure this should return a `Table`? I don't see much use in this as it is now equivalent to `.take_start 1`. Maybe we should return a `Vector` representing the row? I'm not sure, but I think it is worth discussing.

Reply:
This equivalence is true of `Vector` and `Column` as well, so it makes sense to me.

Reply:
I don't think it is.

Let `x : Vector | Column`
`x.take_start 1 : Vector | Column` but `x.first : Any`

Reply:
`take_...` returns a truncated collection, `first` returns an element of the collection. That holds for Vector and Column.

For Table it is less clear because we don't have a `Row` type, so I'm not sure what is the right solution here, but I feel like the current solution to `first` is a bit redundant.